chalk-solve/Cargo.toml

```diff
@@ -27,5 +26,5 @@ chalk-integration = { path = "../chalk-integration" }
 [features]
 default = ["slg-solver", "recursive-solver"]
 
-slg-solver = ["chalk-engine"]
+slg-solver = []
 recursive-solver = []
```

Comment:
I think those features are still used for conditional compilation in chalk-solve, for instance [here](https://github.com/rust-lang/chalk/blob/3aff73e32767bd46ac5e6883cd875bfd56dab2c7/chalk-solve/src/solve.rs#L6-L18).

Reply:
There may be some more refactoring necessary, because the `RecursiveContext` type which is behind the feature flag is used without the feature flag [here](https://github.com/rust-lang/chalk/blob/b7f19b9324db8b4f5fd9cd9f4edde58ad32a2f42/chalk-solve/src/solve.rs#L210-L212) (the crate will fail to compile if the feature flag is turned off).